boot/bootstate20.go

```diff
@@ -145,13 +172,20 @@ func (u20 *bootStateUpdate20) commit() error {
 		}
 	}
 
-	modeenvRewritten := false
+	expectReseal := false
 	// next write the modeenv if it changed
 	if !u20.writeModeenv.deepEqual(u20.modeenv) {
 		if err := u20.writeModeenv.Write(); err != nil {
 			return err
 		}
-		modeenvRewritten = true
+		// reseal might not be needed if the only change in modeenv is
+		// the gadget (if the boot assets change that is detected in
+		// resealKeyToModeenv() and reseal will happen anyway)
+		auxModeEnv := *u20.writeModeenv
+		auxModeEnv.Gadget = u20.modeenv.Gadget
```

Comment:
(nitpick: I would have copied uc20.modeenv, so to use the new Gadget value, not the old)

anyway it might make sense to encapsulate this special comparison logic into a function ResealEqual defined in seal.go

Reply:
Done now